eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+                                                   Map<String, Map<String,String>> studiesSampleNamesEquivalences)
+    {
+        this.sources = sources;
+        this.cellbaseClient = cellbaseWSClient;
+        this.studiesSampleNamesEquivalences = studiesSampleNamesEquivalences;
+        variantContextBuilder = new VariantContextBuilder();
+    }
+
+    public VariantContext transform(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray = getAllelesArray(variant, region);
+
+        Set<Genotype> genotypes = getGenotypes(variant, allelesArray);
+
+        VariantContext variantContext = variantContextBuilder
+                .chr(variant.getChromosome())
+                // TODO: check start and end for indels
+                .start(variant.getStart())
+                .stop(variant.getEnd())
+                .noID()
+                .alleles(allelesArray)
+                .unfiltered()
+                .genotypes(genotypes).make();
+        return variantContext;
+    }
+
+    private String[] getAllelesArray(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray;
+        // if there are indels, we cannot use the normalized alleles, (hts forbids empty alleles) so we have to take them from cellbase
+        if (variant.getReference().isEmpty() || variant.getAlternate().isEmpty()) {
+            String contextNucleotide;
+            if (region != null) {
+                contextNucleotide = getContextNucleotideFromCellbaseCachingRegions(variant, variant.getStart(), region);
+            } else {
+                contextNucleotide = getContextNucleotideFromCellbase(variant, variant.getStart());
+            }
+
+            allelesArray = new String[] {contextNucleotide + variant.getReference(), contextNucleotide + variant.getAlternate()};
+            variant.setEnd(variant.getStart() + allelesArray[0].length() - 1);
+        } else {
+            allelesArray = new String[] {variant.getReference(), variant.getAlternate()};
+        }
+
+        return allelesArray;
+    }
+
+    private String getContextNucleotideFromCellbase(Variant variant, Integer start) throws CellbaseSequenceDownloadError {
+        if (cellbaseClient != null) {
+            String contextNucleotide;
+            int contextNucleotidePosition = start - 1;
+            try {
+                contextNucleotide = cellbaseClient.getSequence(new Region(variant.getChromosome(), contextNucleotidePosition, contextNucleotidePosition));
+                return contextNucleotide;
+            } catch (Exception e) {
+                throw new CellbaseSequenceDownloadError("Error getting from Cellbase sequence for Region " + variant.getChromosome() + ":" +
+                        contextNucleotidePosition + "-" + contextNucleotidePosition, e);
+            }
+        } else {
+            throw new IllegalArgumentException(String.format(
```

Comment:
I would say this would be more an IllegalStateException: "Signals that a method has been invoked at an illegal or inappropriate time."